workspace/exercise.go

```diff
@@ -59,3 +64,54 @@ func (e Exercise) HasMetadata() (bool, error) {
 	}
 	return false, err
 }
+
+// HasLegacyMetadata checks for the presence of a legacy exercise metadata file.
+// If there is no such file, it could also be an unrelated directory.
+func (e Exercise) HasLegacyMetadata() (bool, error) {
+	_, err := os.Lstat(e.LegacyMetadataFilepath())
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// MigrationStatus represents the result of migrating a legacy metadata file.
+type MigrationStatus int
+
+// MigrationStatus
+const (
+	MigrationStatusErrorMkdir MigrationStatus = iota
```

Comment:
It seems a bit odd to have an 'Error' status when we also have an `error` return type. I'm not sure if it's worth preserving for distinguishing these error cases as part of the status or if `error` should be relied on. If we want to ditch `ErrorX` statuses, should they be changed to `...Noop` or something else?

Reply:
I wouldn't worry about having statuses for the errors since (as you mention) we return the error.

I think `MigrationStatusNoop` is fine to return in the error condition, since it would not have actually changed anything if there is an error. The name is fine, too, I think. Could perhaps do `Unchanged` instead of `Noop` but I actually don't feel strongly either way.